tests/actions/IOUTest.js

```diff
@@ -0,0 +1,956 @@
+
+                            // The transactionID on the iou action should match the one from the transactions collection
+                            expect(iouAction.originalMessage.IOUTransactionID).toBe(transactionID);
+
+                            resolve();
+                        },
+                    });
+                }))
+                .then(() => {
+                    fetch.fail();
+                    return fetch.resume();
+                })
+                .then(() => new Promise((resolve) => {
+                    const connectionID = Onyx.connect({
+                        key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${chatReportID}`,
+                        waitForCollectionCallback: true,
+                        callback: (reportActionsForChatReport) => {
+                            Onyx.disconnect(connectionID);
+                            expect(_.size(reportActionsForChatReport)).toBe(2);
+                            iouAction = _.find(reportActionsForChatReport, reportAction => reportAction.actionName === CONST.REPORT.ACTIONS.TYPE.IOU);
+                            expect(iouAction.pendingAction).toBeFalsy();
+                            const errorMessage = _.values(iouAction.errors)[0];
+                            expect(errorMessage).toBe(Localize.translateLocal('iou.error.genericCreateFailureMessage'));
+                            resolve();
+                        },
+                    });
+                }))
+                .then(() => new Promise((resolve) => {
+                    const connectionID = Onyx.connect({
+                        key: `${ONYXKEYS.COLLECTION.TRANSACTION}${transactionID}`,
+                        waitForCollectionCallback: true,
+                        callback: (transaction) => {
+                            Onyx.disconnect(connectionID);
+                            expect(transaction.pendingAction).toBeFalsy();
+                            expect(transaction.errors).toBeTruthy();
+                            expect(_.values(transaction.errors)[0]).toBe(Localize.translateLocal('iou.error.genericCreateFailureMessage'));
+
+                            // Cleanup
+                            fetch.succeed();
+
+                            resolve();
+                        },
+                    });
+                }));
+        });
+    });
+
+    describe('split bill', () => {
+        it('creates and updates new chats and IOUs as needed', () => {
+            /*
+             * Given that:
+             *   - Rory and Carlos have chatted before
+             *   - Rory and Jules have chatted before and have an active IOU report
+             *   - Rory and Vit have never chatted together before
+             *   - There is no existing group chat with the four of them
+             */
+            const amount = 400;
+            const amountInCents = amount * 100;
+            const comment = 'Yes, I am splitting a bill for $4 USD';
```

Comment:
There is an ambiguity here, `amountInCents` is 40000 so that's $400.

Reply:
good catch